Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DPEComponentAdapter.cpp

```diff
@@ -146,4 +152,23 @@ namespace AZ::DocumentPropertyEditor
     {
         ReflectionAdapter::CreateLabel(adapterBuilder, labelText, serializedPath);
     }
+
+    void ComponentAdapter::OnEntityDestroyed(const AZ::EntityId& entityId)
+    {
+        if (entityId == m_entityId)
+        {
+            m_entityId.SetInvalid();
+            AZ::EntitySystemBus::Handler::BusDisconnect();
+        }
+    }
+
+    void ComponentAdapter::OnEntityDeactivated(const AZ::EntityId& entityId)
```

Comment:
I don't think we need OnEntityDeactivated because if the entity gets activated again, nothing will set m_entityId back.